pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -887,24 +888,26 @@ func (og *operationGenerator) GenerateMapVolumeFunc(
 			return volumeToMount.GenerateError("MapVolume failed", fmt.Errorf("Device path of the volume is empty"))
 		}
 
-		// When kubelet is containerized, devicePath may be a symlink at a place unavailable to
-		// kubelet, so evaluate it on the host and expect that it links to a device in /dev,
+		// Map device to global and pod device map path
```

Comment:
Changes here will affect how all block volume devices work including in-tree drivers.  Any changes here should be tested to make sure other drivers are working as before.

Reply:
As for in-tree drivers, I added [e2e tests for block volume](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/in_tree_volumes.go#L51). So, it would be tested there. (Block volume feature was missed to be beta in v1.12, so it won't be automatically tested by CSI, though.)
In addition, it won't be tested with combinations of containerized kubelet, so we might need to find a way to test with containerized kubelet test environment. Let me think how to test it.

As for csi drivers, I'm working on to apply the same testsuites that in-tree drivers do to CSI drivers, [here](https://github.com/kubernetes/kubernetes/issues/68024) and it will make all csi drivers to be tested for block volume. (Actually, no block volume supported CSI drivers there, though.)